scripts/analysis.py

```diff
@@ -356,10 +351,190 @@ def plot_in_out_flux(cursor, facility, influx_bool, title, outputname):
+    pile_dict[facility] = stock_timeseries
+
+    return pile_dict
+
+
+def final_stockpile(cursor, facility):
+    """ get final stockpile in a fuel facility
+
+    Parameters
+    ----------
+    cursor: sqlite cursor
+        sqlite cursor
+    facility: str
+        name of facility
+
+    Returns
+    -------
+    MTHM value of stockpile
+    """
+    cur = cursor
+    stock_dict = collections.OrderedDict({})
+    agentid = get_agent_ids(cur, facility)
+    for agent in agentid:
+        count = 1
+        name = cur.execute('SELECT prototype FROM agententry WHERE agentid = ' + str(agent)).fetchone()
+        print('The Stockpile in ' + name[0] + ' : ')
+        stockpile = cur.execute("""SELECT sum(quantity), inventoryname, qualid
+                                 FROM agentstateinventories
+                                 INNER JOIN resources
+                                 ON resources.resourceid = agentstateinventories.resourceid
+                                 WHERE agentstateinventories.agentid 
+                                 = """ + str(agent) + """ GROUP BY
+                                 inventoryname""").fetchall()
+        for stream in stockpile:
+            masses = cur.execute('SELECT * FROM compositions WHERE qualid = ' + str(stream[2])).fetchall()
+            print('Stream ' + str(count) + ' Total = ' + str(stream[0]) + ' kg')
+            for isotope in masses:
+                print(nucname.name(isotope[2]) + ' = ' + str(isotope[3]*stream[0]) + ' kg')
+            print('\n')
+            count +=1 
+        print('\n')
+    print('\n')
+
+
+def fuel_usage_timeseries(cursor, fuel_list):
+    """ Calculates total fuel usage over time
+
+    Parameters
+    ----------
+    cursor: sqlite cursor
+        sqlite cursor
+    fuel_list: list
+        list of fuel commodity names (eg. uox, mox) to consider in fuel usage.
+
+    Returns
+    -------
+    dict
+        dictionary of different fuels used timeseries
+        keys - fuel
+        vales - timeseries list of fuel amount [kg]
+    """
+
+    cur = cursor
+    fuel_dict = collections.OrderedDict({})
+    for fuel in fuel_list:
+        temp_list = ['"' + fuel + '"']
+        fuel_quantity = cur.execute(exec_string(temp_list, 'commodity',
+                                    'sum(quantity), time')
+                                    + ' GROUP BY time').fetchall()
+        init_year, init_month, duration, timestep = get_sim_time_duration(cur)
+        total_sum = 0
+        quantity_timeseries = []
+
+        try:
+            for i in range(0, duration):
+                fq_array = np.array(fuel_quantity)
+                fq_time = fq_array[:, 1:]
+                quantity = fq_array[fq_time == i]
+                if quantity:
+                    total_sum += fq_array[fq_time == i][0]
+                quantity_timeseries.append(total_sum)
+            fuel_dict[fuel] = quantity_timeseries
+        except:
+            print('No Such Fuel Called ' + fuel)
```

Comment:
It depends on your answer to my question. "do you really want this program to continue if the fuel type is not found?"